tests/functional/adapter/incremental/test_incremental_on_schema_change.py

```diff
@@ -251,3 +251,59 @@ def test_run_incremental_sync_all_columns_time_ingestion_partitioning(self, proj
         compare_source = "incremental_sync_all_columns_time_ingestion_partitioning"
         compare_target = "incremental_sync_all_columns_time_ingestion_partitioning_target"
         self.run_twice_and_assert(select, compare_source, compare_target, project)
+
+
+_MODELS__SRC_ARTISTS = """
+{{
+    config(
+        materialized='table',
+    )
+}}
+{% if var("version", 0) == 0 %}
+    select {{ dbt.current_timestamp() }} as inserted_at, 'john' as name
+{% else %}
+    -- add a non-zero version to the end of the command to get a different version:
+    -- --vars "{'version': 1}"
+    select {{ dbt.current_timestamp() }} as inserted_at, 'engineer' as Job,
+{% endif %}
+"""
+
+_MODELS__DIM_ARTISTS = """
+{{
+    config(
+        materialized='incremental',
+        on_schema_change='append_new_columns',
+    )
+}}
+select  * from {{ ref("src_artists") }}
+"""
+
+
+class BaseBigQueryIncrementalCaseSenstivityOnSchemaChange:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "src_artists.sql": _MODELS__SRC_ARTISTS,
+            "dim_artists.sql": _MODELS__DIM_ARTISTS,
+        }
+
+    def test_incremental_case_sensitivity(self, project):
```

Comment:
I could be missing it, but this looks like the same test as in the base. Can we inherit from the new test case and just overwrite `models`? Also, this should just be a test class. We don't need the base class and the test class since this is in the concrete adapter.

Reply:
I had pulled the dbt-adapter ones into its own base case and test class so that the location I had them originally as part of `BaseIncrementalOnSchemaChangeSetup` but didn't want to always pull it in since bigquery requires its own default version